rs/sns/governance/src/governance.rs

```diff
@@ -477,16 +477,19 @@ impl GovernanceProto {
     {
         let deployed_version = self.deployed_version_or_err()?;
 
-        let upgrade_steps = {
-            let cached_upgrade_steps = self.cached_upgrade_steps_or_err()?;
-            cached_upgrade_steps.take_from(&deployed_version)?
-        };
+        let cached_upgrade_steps = self.cached_upgrade_steps_or_err()?;
 
-        if upgrade_steps.is_empty() {
-            return Err(
-                "Cannot advance SNS target version: there are no pending upgrades.".to_string(),
-            );
-        }
+        let upgrade_steps = cached_upgrade_steps.take_from(&deployed_version);
+        let upgrade_steps = match upgrade_steps {
+            Ok(upgrade_steps) if !upgrade_steps.is_empty() => upgrade_steps,
```

Comment:
NCR; we can do this eventually

```suggestion
            Ok(upgrade_steps) if upgrade_steps.has_pending_upgrades() => upgrade_steps,
```

Reply:
Can't wait!